Kentico.Kontent.Management.Tests/EndpointUrlBuilderTests.cs

```diff
@@ -205,6 +206,88 @@ public void BuildAssetsUrlFromExternalId_WithGivenAssetId_ReturnsExpectedUrl()
+        {
+            var assetId = Guid.NewGuid();
+            var renditionId = Guid.NewGuid();
+            var expectedResult = $"{ENDPOINT}/projects/{PROJECT_ID}/assets/{assetId}/renditions/{renditionId}";
+            var actualResult = _builder.BuildAssetRenditionsUrl(new AssetRenditionIdentifier(Reference.ById(assetId), Reference.ById(renditionId)));
+
+            Assert.Equal(expectedResult, actualResult);
+        }
+        
+        [Fact]
+        public void BuildAssetRenditionsUrlFromAssetIdAndRenditionExternalId_WithGivenAssetAndRenditionIds_ReturnsExpectedUrl()
+        {
+            var assetId = Guid.NewGuid();
+            var renditionExternalId = "customized-brewer";
+            var expectedResult = $"{ENDPOINT}/projects/{PROJECT_ID}/assets/{assetId}/renditions/external-id/{renditionExternalId}";
+            var actualResult = _builder.BuildAssetRenditionsUrl(new AssetRenditionIdentifier(Reference.ById(assetId), Reference.ByExternalId(renditionExternalId)));
+
+            Assert.Equal(expectedResult, actualResult);
+        }
+        
+        [Fact]
+        public void BuildAssetRenditionsUrlFromAssetExternalIdAndRenditionId_WithGivenAssetAndRenditionIds_ReturnsExpectedUrl()
+        {
+            var assetExternalId = "which-brewing-fits-you";
+            var renditionId = Guid.NewGuid();
+            var expectedResult = $"{ENDPOINT}/projects/{PROJECT_ID}/assets/external-id/{assetExternalId}/renditions/{renditionId}";
+            var actualResult = _builder.BuildAssetRenditionsUrl(new AssetRenditionIdentifier(Reference.ByExternalId(assetExternalId), Reference.ById(renditionId)));
+
+            Assert.Equal(expectedResult, actualResult);
+        }
+        
+        [Fact]
+        public void BuildAssetRenditionsUrlFromAssetExternalIdAndRenditionExternalId_WithGivenAssetAndRenditionIds_ReturnsExpectedUrl()
+        {
+            var assetExternalId = "which-brewing-fits-you";
+            var renditionExternalId = "customized-brewer";
+            var expectedResult = $"{ENDPOINT}/projects/{PROJECT_ID}/assets/external-id/{assetExternalId}/renditions/external-id/{renditionExternalId}";
+            var actualResult = _builder.BuildAssetRenditionsUrl(new AssetRenditionIdentifier(Reference.ByExternalId(assetExternalId), Reference.ByExternalId(renditionExternalId)));
+
+            Assert.Equal(expectedResult, actualResult);
+        }
+        
+        [Fact]
+        public void BuildAssetRenditionsUrlFromAssetCodenameAndRenditionId_AssetDoesNotSupportCodename_Throws()
+        {
+            _builder.Invoking(x => x.BuildAssetRenditionsUrl(new AssetRenditionIdentifier(Reference.ByCodename("not-supported"), Reference.ById(Guid.NewGuid()))))
+                .Should().ThrowExactly<InvalidOperationException>();
+        }
+        
+        [Fact]
+        public void BuildAssetRenditionsUrlFromAssetIdAndRenditionCodename_RenditionDoesNotSupportCodename_Throws()
```

Comment:
I don't see such tests for other methods either. Methods for variant URLs (such as `LanguageVariantIdentifier`) would fail with a `NullPointerException` as well. Are you trying to make me write the test in order to go and fix the code as well? :D

Reply:
I think we are missing a lot of tests for `EndpointUrlBuilder` and we have to start with something :)